platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+3. **Bilinear Pairings API**: Layer of abstraction to be included at compilation time providing the cryptography primitives
+   and the arithmetic operations for working with a specific EC curve and the underlying Groups, Fields, and Pairings
+   and minimizes the impact of changing or including different curve implementations.
+4. **Bilinear Pairings Impl**: This is an implementation of the Bilinear Pairings API that will be loaded at runtime using Java’s SPI.
+   Multiple implementations can be provided to support different types of curves. It allows changing or switching dependencies.
+5. **Native Support Lib**: Provides a set of generic functions loading native libraries in different system architectures when packaged in a jar
+   using a predefined organization so they can be accessed with JNI.
+6. **[Arkworks](https://github.com/arkworks-rs)**: A Rust ecosystem for cryptography. Our implementation uses it as the library responsible for elliptic curve cryptography.
+7. **EC-Key Utils** is a utility module that enables the node operator to generate a bootstrapping public/private key pair.
+
+### Module organization and repositories
+![img_6.png](img_6.svg)
+1. **hedera-cryptography**: This is a separate repository for hosting cryptography-related libraries.
+   It is necessary to facilitate our build process, which includes Rust libraries. It also provides independent release cycles between consensus node code and block node code.
+2. **swirlds-native-support**: Gradle module that enables loading into memory compiled native libraries so they can be used with JNI.
+3. **swirlds-crypto-tss**: Gradle module for the TSS Library.
+   This library's only client is the consensus node, so it will be in the hedera-services repository, under the `platform-sdk` folder.
+4. **swirlds-crypto-signatures**: Gradle module for the Bilinear Pairings Signature Library.
+5. **swirlds-crypto-pairings-api**: Gradle module for the Bilinear Pairings API. Minimizes the impact of adding or removing implementations.
+6. **swirlds-crypto-altbn128**: Gradle module that will implement the Bilinear Pairings API using alt-bn128 elliptic curve.
+   That curve has been chosen due to EVM support. The arkworks rust library will provide the underlying cryptography implementation.
+   The module will include Java and Rust codes that will be compiled for all possible system architectures and distributed in a jar under a predefined structure.
+
+### Handling of multilanguage modules and native code distribution
+The code provided by this solution will include multilanguage modules (rust + java). Rust code needs to be compiled into binary libraries and accessed through JNI.
+Given the non-GA state of Project Panama, this proposal does not consider it, but it has not been discarded for future change proposals.
+
+There are two possible ways of loading libraries and accessing them through JNI
+1) Libraries are installed in the SO and referenced through the classpath.
```

Comment:
The `SO` term should probably be defined.
This is what I _think_ was intended (but I'm guessing a little here).
```suggestion
1) Libraries are installed on the system as shared object(SO) libraries and found via the classpath and system library path.
```